bahmnicore-api/src/test/java/org/bahmni/module/bahmnicore/forms2/mapper/FormDetailsMapperTest.java

```diff
@@ -0,0 +1,238 @@
+package org.bahmni.module.bahmnicore.forms2.mapper;
+
+import org.bahmni.module.bahmnicore.forms2.contract.form.FormType;
+import org.bahmni.module.bahmnicore.forms2.contract.form.data.FormDetails;
+import org.bahmni.module.bahmnicore.forms2.util.FormUtil;
+import org.bahmni.module.bahmnicore.model.Provider;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.internal.verification.VerificationModeFactory;
+import org.openmrs.Concept;
+import org.openmrs.ConceptName;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.PersonName;
+import org.openmrs.User;
+import org.openmrs.Visit;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.reflect.Whitebox;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashSet;
+
+import static java.util.Collections.singletonList;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+
+@PrepareForTest({FormUtil.class, FormType.class})
+@RunWith(PowerMockRunner.class)
+public class FormDetailsMapperTest {
+
+    private String formFieldPath = "FormName.2/1-0";
+    private String encounterUuid = "encounter-Uuid";
+    private String visitUuid = "visitUuid";
+    private String providerName = "Super Man";
+    private String providerUuid = "provider-uuid";
+    private String formName = "formName";
+    private int formVersion = 2;
+    private Date encounterDateTime = new Date();
+    private Date visitStartDateTime = new Date();
+
+    private Obs obs = mock(Obs.class);
+    private Encounter encounter = mock(Encounter.class);
+    private Visit visit = mock(Visit.class);
+    private User anotherCreator = mock(User.class);
+    private PersonName personName = mock(PersonName.class);
+
+    @Before
+    public void setUp() {
+
+        mockStatic(FormUtil.class);
```

Comment:
Why do you want to mock FormUtil? Its a simple class without any dependencies, and only does simple string parsing. Its better that its part of the executing code.

Reply:
In a unit level test, I prefer to mock because of 

- future code changes coming in FormUtil won't disturb the test cases in FormDetailsMapperTest except for FormUtil tests.

- Moreover, I see it as a dependency to the FormDetailsmapperTest.